misc/test_bit_array_vectorization.py

```diff
@@ -0,0 +1,59 @@
+import taichi as ti
+
+ti.init(debug=True, cfg_optimization=False, kernel_profiler=True)
+
+vectorize = False
+
+ci1 = ti.type_factory_.get_custom_int_type(1, False)
+
+x = ti.field(dtype=ci1)
+y = ti.field(dtype=ci1)
+
+N = 4096
+block_size = 4
+bits = 32
+boundary_offset = 64
+
+block = ti.root.pointer(ti.ij, (block_size, block_size))
+block.dense(ti.ij, (N // block_size, N // (bits * block_size)))._bit_array(
+    ti.j, bits, num_bits=bits).place(x)
+block.dense(ti.ij, (N // block_size, N // (bits * block_size)))._bit_array(
+    ti.j, bits, num_bits=bits).place(y)
+
+
+@ti.kernel
+def init():
+    for i, j in ti.ndrange((boundary_offset, N - boundary_offset),
+                           (boundary_offset, N - boundary_offset)):
+        x[i, j] = (N * i + j) % 2
```

Comment:
This test is not strong enough to be indicative. Maybe use `ti.random(dtype=ti.i32) % 2` in `init` and `x[i, j] == y[i, j]` in `verify`?